std/experimental/allocator/package.d

```diff
@@ -1099,7 +1099,11 @@ void dispose(A, T)(auto ref A alloc, T p)
 if (is(T == class) || is(T == interface))
 {
     if (!p) return;
-    auto support = (cast(void*) p)[0 .. typeid(p).initializer.length];
+    static if (is(T==interface))
+        auto ob = cast(Object) p;
```

Comment:
I guess if `cast(Object)` fails then try casting to `IUknown`? I'm not really sure how to deal with this case.